### ISSUE TITLE
Progress with React.JS

### ISSUE THREAD

**Issue by berkaey** [Lepozepo/S3#131]
Hey,

I've been trying to manage progress indicator works with react but nothing is triggering.

I tried this
```
meteorContainer =  (props)->
	return { files: S3.collection.find().fetch() }
module.exports = React.createFactory createContainer meteorContainer , View
```

this
```
meteorContainer =  (props)->
        Meteor.subscribe("S3.collection")
	return { files: S3.collection.find().fetch() }
module.exports = React.createFactory createContainer meteorContainer , View
```
and event with https://github.com/kadirahq/react-komposer

nothing happens while uploading.

Do I need to do anything else to listen progesss ?

**Comment by Lepozepo** [comment]
Hey! You definitely don't need a subscribe function. That collection is client-side only. Remember the collection is reactive so you need to put it inside a createContainer instead for it to change when it's dependencies change. Have you tried that?

**Comment by berkaey** [comment]
yeap it is actually inside createContainer

let me rephrase in vanilla JS, sometimes coffee is hard to read.

File.js/coffee
```
{createContainer} = require("meteor/react-meteor-data")
View = React.createClass
 getFiles:(event)->

...
...
...

module.exports = React.createFactory(createContainer(function(props){
 console.log('hello') // this triggers only for first time.
 return { files: S3.collection.find().fetch() }
},View))
```
btw, normal subscriptions just works it triggers that createContainer.

**Comment by Lepozepo** [comment]
Mmm, that should work, you should see files in the props object of your react component changing as the uploaded changes. I'll have a look as soon as I can.

**Comment by berkaey** [comment]
Thanks, man. Great work btw. ;)
